tests/unit/sources/test_git_source.py

```diff
@@ -156,16 +156,25 @@ def test_pull_tag(self, fake_run, new_dir):
             ]
         )
 
-    def test_pull_commit(self, fake_run, new_dir):
+    def test_pull_commit(self, fake_check_output, fake_run, new_dir):
+        fake_check_output.return_value = "2514f9533ec9b45d07883e10a561b248497a8e3c"
         git = GitSource(
             "git://my-source",
             Path("source_dir"),
             cache_dir=new_dir,
-            source_commit="2514f9533ec9b45d07883e10a561b248497a8e3c",
+            source_commit="2514f9533e",
```

Comment:
these feel like *new* test cases, in addition to the existing ones

Reply:
As-is, I decided not to add another test because the unit tests are fully mocked and it doesn't matter in the slightest what I pass to `source-commit`. I made this change to ensure the commands used the return value from `fake_check_output.return_value = "2514f9533ec9b45d07883e10a561b248497a8e3c"` and not what was passed to `source_commit`.

If I add an early return for full-length commits like you suggested, then I agree adding a new test would exercise additional code.